README.MD

```diff
@@ -47,52 +52,117 @@ Certainly! The installation of the `olm` library may vary depending on the opera
+
+2. **Install libolm:**
    ```bash
    brew install libolm
    ```
 
-#### Verify Installation
+## Build
 
-After installing `olm`, you can verify its installation by checking the library version:
+Use the provided build script to build the application. This will create a binary in the `./build` directory.
 
 ```bash
-olm --version
+./scripts/build.sh
 ```
 
-This command should print the installed `olm` version.
+## Run
+
+### Configuration
+
+Use the provided example configuration files for suppliers
+(`camino-messenger-bot-supplier.yaml.example`) and distributors
+(`camino-messenger-bot-distributor.yaml.example`).
+
+Copy them accordingly and edit fields, mainly the private keys. Most of the default
+values should be fine for testing.
+
+> [!IMPORTANT]
+> Keep in mind that the addresses of `evm_private_key` in the config file needs to
+> have funds on C-Chain to pay for the transaction fees. (If you are using **MintRequest**)
+
+### Start the Application
+
+Start the bot using the related config file depending on your need.
+
+**Supplier Bot:**
+
+```bash
+./build/bot --config camino-messenger-bot-supplier.yaml
+```
+
+**Distributor Bot:**
+
+```bash
+./build/bot --config camino-messenger-bot-distributor.yaml
+```
 
-Note: The provided instructions assume a certain level of system familiarity. Adjustments might be necessary based on your specific setup or if you encounter any issues during the installation process.
+> [!TIP]
+> For testing and development, you can also start two bots, one supplier and one distributor.
+
+Feel free to explore and customize the application based on your specific requirements. If you encounter any issues or have questions, please refer to the documentation or open an issue in the repository.
+
+## Running both Supplier and Distributor at the same time
+
+For development proposes, you may want to run both Supplier and Distributor bot at the same time.
 
 ### Build
 
 ```bash
-go build -o bot cmd/camino-messenger-bot/main.go
+./scripts/build_both.sh
 ```
 
+to build bots to different folders.
+
 ### Create Configuration
 
-Create a configuration YAML file based on the provided example "camino-messenger-bot.yaml.example".
+Use `camino-messenger-bot-distributor.yaml.example` and `camino-messenger-bot-supplier.yaml.example` files to create configuration YAML files (e.g. `camino-messenger-bot-distributor.yaml` and `camino-messenger-bot-supplier.yaml`).
 
 ### Start the Application
 
+To run the bots, in 2 separate terminals run:
+
 ```bash
-./bot --config camino-messenger-bot.yaml
+./build/distributor/bot --config camino-messenger-bot-distributor.yaml
 ```
 
-Feel free to explore and customize the application based on your specific requirements. If you encounter any issues or have questions, please refer to the documentation or open an issue in the repository.
+```bash
+./build/supplier/bot --config camino-messenger-bot-supplier.yaml
+```
+
+Or configure the launching file of your preferred IDE to do the same.
+
```

Comment:
I believe we do not need this whole section (running both bots at the same time), wdyt @VjeraTurk?